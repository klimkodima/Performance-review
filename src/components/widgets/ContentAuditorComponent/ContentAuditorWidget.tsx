import { FC } from 'react';
import { useTranslation } from 'react-i18next';
import { ContentImage } from '../../common';
import ContentRow from './ContentRow';
import { auditorInfo } from './data';
import './ContentAuditorWidget.scss';
import imageIcon from './img/01_Auditor_icon.png';

const ContentAuditorWidget: FC = () => {
  const { t } = useTranslation('common', { keyPrefix: 'contentAuditor' });
  return (
    <div className='widget' data-testid='widget'>
      <div className='widget__content content' data-testid='content'>
        <h3 className='content__header' data-testid='content-header'>
          {t('Title')}
        </h3>
        <div className='content__block' data-testid='content-block'>
          {Object.entries(auditorInfo)
            .filter((item) => item[0] !== 'photo')
            .map((item) => (
              <ContentRow
                key={item[0]}
                classesContainer='content__row'
                classesLabel='label'
                classesText='text'
                label={item[0]}
                text={item[1]}
              />
            ))}
        </div>
        {auditorInfo.photo.trim().length ? (
          <ContentImage
            classes='content__img'
            src={auditorInfo.photo}
            alt={auditorInfo.Name}
          />
        ) : (
          <ContentImage
            classes='content__img'
            src={imageIcon}
            alt='user icon'
          />
        )}
      </div>
    </div>
  );
};

export default ContentAuditorWidget;
