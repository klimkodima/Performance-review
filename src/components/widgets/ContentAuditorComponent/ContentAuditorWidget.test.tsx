import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import ContentAuditorWidget from './ContentAuditorWidget';

describe('My Connected React-Redux Component', () => {
  test('Matches snapshot', () => {
    const tree = renderer.create(<ContentAuditorWidget />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders content-auditor-component', () => {
    render(<ContentAuditorWidget />);
    const widget = screen.getByTestId('widget');
    const content = screen.getByTestId('content');
    const contentBlock = screen.getByTestId('content-block');
    const contentHeader = screen.getByTestId('content-header');
    expect(widget).toBeInTheDocument();
    expect(content).toBeInTheDocument();
    expect(contentBlock).toBeInTheDocument();
    expect(contentHeader).toBeInTheDocument();
  });

  test('renders image block', () => {
    render(<ContentAuditorWidget />);
    const imageBlock = screen.getByTestId('image-block');
    const image = screen.getByTestId('image');
    expect(imageBlock).toBeInTheDocument();
    expect(image).toBeInTheDocument();
  });

  test('renders content row block 5 times', () => {
    render(<ContentAuditorWidget />);
    const container = screen.getAllByTestId('container');
    const label = screen.getAllByTestId('label');
    const text = screen.getAllByTestId('text');
    expect(container.length).toBe(5);
    expect(label.length).toBe(5);
    expect(text.length).toBe(5);
  });

  test('renders classes of components accordingly to layout', () => {
    render(<ContentAuditorWidget />);
    const widget = screen.getByTestId('widget');
    const content = screen.getByTestId('content');
    const contentBlock = screen.getByTestId('content-block');
    const contentHeader = screen.getByTestId('content-header');
    expect(widget).toHaveClass('widget');
    expect(content).toHaveClass('widget__content content');
    expect(contentBlock).toHaveClass('content__block');
    expect(contentHeader).toHaveClass('content__header');
  });

  test('image has alt text', () => {
    render(<ContentAuditorWidget />);
    const image = screen.getByTestId('image');
    expect(image).toHaveAttribute('alt');
  });
});
