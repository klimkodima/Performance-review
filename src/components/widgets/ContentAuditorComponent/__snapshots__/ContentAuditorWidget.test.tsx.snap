// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`My Connected React-Redux Component Matches snapshot 1`] = `
<div
  className="widget"
  data-testid="widget"
>
  <div
    className="widget__content content"
    data-testid="content"
  >
    <h3
      className="content__header"
      data-testid="content-header"
    >
      Title
    </h3>
    <div
      className="content__block"
      data-testid="content-block"
    >
      <div
        className="content__row"
        data-testid="container"
      >
        <span
          className="label"
          data-testid="label"
        >
          Name
          :
        </span>
        <span
          className="text"
          data-testid="text"
        >
          Elena Elenova
        </span>
      </div>
      <div
        className="content__row"
        data-testid="container"
      >
        <span
          className="label"
          data-testid="label"
        >
          Team
          :
        </span>
        <span
          className="text"
          data-testid="text"
        >
          X-Rays
        </span>
      </div>
      <div
        className="content__row"
        data-testid="container"
      >
        <span
          className="label"
          data-testid="label"
        >
          Level
          :
        </span>
        <span
          className="text"
          data-testid="text"
        >
          Team Lead
        </span>
      </div>
      <div
        className="content__row"
        data-testid="container"
      >
        <span
          className="label"
          data-testid="label"
        >
          Team Lead
          :
        </span>
        <span
          className="text"
          data-testid="text"
        >
          Elena Elenova
        </span>
      </div>
      <div
        className="content__row"
        data-testid="container"
      >
        <span
          className="label"
          data-testid="label"
        >
          Experience in PV
          :
        </span>
        <span
          className="text"
          data-testid="text"
        >
          3 years 5 months
        </span>
      </div>
    </div>
    <div
      className="content__img"
      data-testid="image-block"
    >
      <img
        alt="user icon"
        data-testid="image"
        src="01_Auditor_icon.png"
      />
    </div>
  </div>
</div>
`;
